src/lib.rs

```diff
@@ -14,17 +14,66 @@ mod value;
 pub use value::Value;
 pub use error::CalcError;
 
+/// Evalulates a regular mathematical expression.
 pub fn eval(input: &str) -> Result<Value, CalcError> {
     let mut env = parse::DefaultEnvironment;
     token::tokenize(input).and_then(|x| parse::parse(&x, &mut env))
 }
 
+/// Evalulates a regular mathematical expression, with extra environment
+/// variables.
 pub fn eval_with_env<E>(input: &str, env: &mut E) -> Result<Value, CalcError>
     where E: parse::Environment
 {
     token::tokenize(input).and_then(|x| parse::parse(&x, env))
 }
 
+/// Evalulates mathematical expressions that are written in Polish Notation.
+///
+/// Polish Notation defines that a string of operators are given at the
+/// beginning of the
+/// expression, as prefixes, and are followed by a string of numbers to apply
+/// each operation
+/// to. Polish Notation enables writing mathematical expressions that don't
+/// require grouping
+/// via parenthesis. It's also referred to as Prefix Notation, or Normal Polish
+/// Notation (NPN).
+///
+/// # Examples
+///
+/// - `+ * 3 4 5` is equivalent to `3 * 4 + 5`
+/// - `+ / * 5 3 2 * + 1 3 5` is equivalent to `((5 * 3) / 2) + ((1 + 3) * 5)`
```

Comment:
These could be doc comments, something like:
```
/// ```
/// assert_eq!(eval_polish(FOO), eval(FOO));
/// ```
```